### PR TITLE
Introduce langlib `count`, `first` and `last` functions

### LINKED ISSUE #39658: [Improvement]: Introduce langlib function `count`, `first` and `last`
### Description

Please check https://github.com/ballerina-platform/ballerina-spec/issues/1135#issuecomment-1200665146

### Describe your problem(s)

_No response_

### Describe your solution(s)

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose

Fixes #39658

### CODE REVIEW

File: langlib/lang.value/src/main/ballerina/value.bal [hunk 1]
```diff
@@ -423,3 +423,46 @@ public isolated function mergeJson(json j1, json j2) returns json|error = @java:
...
+# Returns the first argument.
+#
+# ```ballerina
+# value:first(1, 2, 3) ⇒ 1
+# ```
+# 
+# + v - first argument
+# + vs - rest of the arguments
+# + return - first argument
+public isolated function first(Type v, (any|error)... vs) returns Type {
```
Comment (gimantha) [inline langlib/lang.value/src/main/ballerina/value.bal:452]: We need to return () when there are no arguments in the min/amx/last/first/avg right?, Shall we add tests for those cases as well?

Comment (KavinduZoysa) [reply langlib/lang.value/src/main/ballerina/value.bal:452]: Please check this comment. https://github.com/ballerina-platform/ballerina-spec/issues/1135#issuecomment-1200665146

Among `last/first/count`, only the `count` can have no args. In that case, also we have to return 0. Check the test [case](https://github.com/ballerina-platform/ballerina-lang/pull/39667/files#diff-875e3c55bfca3d68e3f568d1ba716a410110ba66aa4fa71a5e1d8fc2f019e032R3067). Therefore There is not a case of returning `()`.

File: langlib/lang.value/src/main/java/org/ballerinalang/langlib/value/Count.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.langlib.value;
+
+/**
+ * Extern function lang.values:count.
+ *
+ * @since 2201.5.0
```
[the remark this answers is not in the document]
Comment (KavinduZoysa) [reply langlib/lang.value/src/main/java/org/ballerinalang/langlib/value/Count.java]: Fixed

File: langlib/lang.value/src/main/ballerina/value.bal (hunk 1 above)
[the remark this answers is not in the document]
Comment (KavinduZoysa) [reply langlib/lang.value/src/main/ballerina/value.bal]: Fixed

Comment (gimantha) [reply langlib/lang.value/src/main/ballerina/value.bal]: This comment was posted after that: https://github.com/ballerina-platform/ballerina-spec/issues/1135#issuecomment-1200678526

**Review by LakshanWeerasinghe — APPROVED** [review]

**Pull request merged**